File: 007_Reverse_Integer.cpp
class Solution {
public:
  int reverse(int x) {
    int sign = x>0 ? 1 : -1;
    x = std::abs(x);
    long res(0);
    while (x) {
      res = res * 10 + x % 10;
      x/=10;
    }
    res *= sign;
    return res>INT_MAX||res<INT_MIN? 0 : res;
  }
};

File: java/007_Reverse_Integer.java
public class Solution {
    public int reverse(int x) {
        int sign = x > 0 ? 1 : -1;
        x = Math.abs(x);
        long reval = 0;
        while (x != 0) {
            reval = reval * 10 + x % 10;
            x /= 10;
        }
        reval *= sign;
        return reval > Integer.MAX_VALUE || reval < Integer.MIN_VALUE ? 0 : (int)reval;
    }
}
